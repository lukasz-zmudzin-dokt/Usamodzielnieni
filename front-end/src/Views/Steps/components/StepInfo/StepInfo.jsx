import React from 'react';
import { Card, Button } from 'react-bootstrap';

const StepInfo = ({ step }) => {
    return (
        <Card className="stepInfo">
            <Card.Body>
                <p>{step.value}</p>
                {step.next.map(nextStep => <Button variant="primary">{nextStep}</Button>)}
            </Card.Body>
        </Card>
    )
}

export default StepInfo
