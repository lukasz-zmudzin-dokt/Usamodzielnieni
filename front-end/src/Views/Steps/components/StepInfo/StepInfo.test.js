import React from "react";
import { render, fireEvent } from "@testing-library/react";
import StepInfo from "./StepInfo";

describe("StepInfo", () => {
  let props;

  beforeEach(() => {
    props = {
      step: {
        value: "Opis",
        next: [
          { id: "S001", choiceName: "Tak" },
          { id: "S002", choiceName: "Nie" },
        ],
      },
      setCurrent: jest.fn(),
    };
  });

  it("should render without crashing", () => {
    const { container } = render(<StepInfo {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render without buttons when next is undefined", () => {
    props.next = undefined;
    const { container } = render(<StepInfo {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should set next step as current when button is clicked", () => {
    const { getByText } = render(<StepInfo {...props} />);

    fireEvent.click(getByText("Nie"));

    expect(props.setCurrent).toHaveBeenCalledWith("S002");
  });
});
