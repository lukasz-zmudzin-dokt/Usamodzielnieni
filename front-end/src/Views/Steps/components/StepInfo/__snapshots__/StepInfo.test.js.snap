// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepInfo should render without buttons when next is undefined 1`] = `
<div>
  <div
    class="stepInfo card"
  >
    <div
      class="card-body"
    >
      <p>
        Opis
      </p>
      <button
        class="mr-2 btn btn-primary"
        type="button"
      >
        Tak
      </button>
      <button
        class="mr-2 btn btn-primary"
        type="button"
      >
        Nie
      </button>
    </div>
  </div>
</div>
`;

exports[`StepInfo should render without crashing 1`] = `
<div>
  <div
    class="stepInfo card"
  >
    <div
      class="card-body"
    >
      <p>
        Opis
      </p>
      <button
        class="mr-2 btn btn-primary"
        type="button"
      >
        Tak
      </button>
      <button
        class="mr-2 btn btn-primary"
        type="button"
      >
        Nie
      </button>
    </div>
  </div>
</div>
`;
